src/main/scala/lms/thirdparty/mpi.scala

```diff
@@ -0,0 +1,76 @@
+package lms.thirdparty
+
+import lms.core._
+import lms.util._
+import lms.core.stub._
+import lms.core.Backend._
+import lms.core.virtualize
+import lms.core.utils.time
+import lms.macros.SourceContext
+
+trait MPIOps extends lms.collection.PointerOps { b: Base =>
+  /* LMS support for MPI library */
+
+  def mpi_init(): Rep[Unit] = Wrap[Unit](Adapter.g.reflectWrite("mpi-init")(Adapter.CTRL))
```

Comment:
Is it something you witnessed or you are afraid that would happen? The code isn't moving around freely, so we know when it will move. What mechanism would make that statement move? If I write code that does:

```
def main() = {
   mpi_init()
   // ...
}
```

There is absolutely now reason that this statement won't be first. The code motion mechanisms are:
- inlining: not going to happen with that one.
- move non effectful statement in if block: not going to happen
- take non effectful statement out of a while/lambda block: not going to happen.

Reply:
Thanks Greg. Your analysis assures that the code will be at the top of the snippet. 

But do we need a `stop-of-the-world` effect anyway? If so, how do you suggest we do it?